src/mlpack/tests/cosine_tree_test.cpp

```diff
@@ -105,14 +107,58 @@ BOOST_AUTO_TEST_CASE(CosineNodeCosineSplit)
 
       // Check if the columns assigned to the children agree with the splitting
       // condition.
-      double cosineMax = arma::max(cosines % (cosines < 1));
+      // Due to miscalculations cosineMax calculated at CosineNodeSplit
+      // may differ from cosineMax below
+      double cosineMax = arma::max(cosines % (cosines < 1.0 + precision));
       double cosineMin = arma::min(cosines);
-
-      for (i = 0; i < leftIndices.size(); i++)
-        BOOST_CHECK_LT(cosineMax - cosines(i), cosines(i) - cosineMin);
-
-      for (j = 0, k = i; j < rightIndices.size(); j++, k++)
-        BOOST_CHECK_GT(cosineMax - cosines(k), cosines(k) - cosineMin);
+      // If max(cosines) is close to 1.0 cosineMax and cosineMax2 may
+      // differ significantly
+      double cosineMax2 = arma::max(cosines % (cosines < 1.0 - precision));
+
+
+      if(std::fabs(cosineMax - cosineMax2) < precision)
+      {
+        // Check with some precision
+	    for (i = 0; i < leftIndices.size(); i++)
+          BOOST_CHECK_LT(cosineMax - cosines(i), cosines(i) - cosineMin + precision);
```

Comment:
Hm, these should be `BOOST_REQUIRE_LT` instead of `BOOST_CHECK_LT` like the rest of the mlpack tests, I'm not sure why the original code used `CHECK` instead of `REQUIRE`.  I'll go ahead and change this after the merge.